app/lib/notifier.rb

```diff
@@ -23,17 +23,18 @@ def guiders_to_notify(appointment)
   end
 
   def notify_customer(appointment)
-    if appointment_time_changed?(appointment)
-      AppointmentMailer.updated(appointment).deliver_later
+    if appointment_details_changed?(appointment)
+      AppointmentUpdatedNotificationJob.perform_later(appointment)
     elsif appointment_cancelled?(appointment)
-      AppointmentMailer.cancelled(appointment).deliver_later
+      AppointmentCancelledNotificationJob.perform_later(appointment)
     elsif appointment_missed?(appointment)
-      AppointmentMailer.missed(appointment).deliver_later
+      AppointmentMissedNotificationJob.perform_later(appointment)
     end
   end
 
-  def appointment_time_changed?(appointment)
-    appointment.previous_changes.slice('start_at', 'end_at').present?
+  def appointment_details_changed?(appointment)
+    appointment.previous_changes.any? &&
```

Comment:
And do the same thing with `appointment_cancelled?` and `appointment_missed?` for consistency?